client/components/count/index.jsx

```diff
@@ -2,22 +2,22 @@
  * External dependencies
  */
 import React from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
+import { localize } from 'i18n-calypso';
+import { identity, omit } from 'lodash';
 
-export default React.createClass( {
+const Count = ( { count, numberFormat, ...inheritProps } ) => (
+	<span className="count" { ...omit( inheritProps, [ 'translate', 'moment' ] ) }>
+		{ numberFormat( count ) }
+	</span>
+);
 
-	displayName: 'Count',
+Count.propTypes = {
+	count: React.PropTypes.number.isRequired,
+	numberFormat: React.PropTypes.func
+};
 
-	mixins: [ PureRenderMixin ],
+Count.defaultProps = {
+	numberFormat: identity
```

Comment:
So it looks like it works without defaultProps.

`identity` is not enough for tests. It does not do any formatting and does return `Number` instead of `String` what tests expect.  Wondering if we actually need most those tests? They do not check `Count` at all, they just check implementation of `numberFormat`, I have made mocked `numberFormat` that allows tests to pass. We can either use this "more" elaborate mocking or ditch the tests.